fix(ivy): do not inject attributes with namespace

cexbrayat opened the pull request:
When injecting with `@Attribute`, namespaced attributes should not match (in order to have feature parity with View Engine).

This PR resolves FW-1137

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: FW-1137


## What is the new behavior?

DI now skips namespaced attributes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

kara commented on packages/core/test/render3/di_spec.ts:
```diff
@@ -2154,6 +2154,36 @@ describe('di', () => {
       expect(nonExist).toBeNull();
     });
 
+    it('should not inject attributes with namespace', () => {
```
We're trying to move away from writing generated code by hand, so new tests should be written with TestBed in the `core/test/acceptance` folder. There is another `di_spec` there - can you convert this test?

cexbrayat replied:
@kara I gave it a try, but even a simple test to inject an attribute like the following is failing:

```typescript
it('should inject attribute', () => {
  @Component({
    selector: 'my-comp',
    template: '<div exist="existValue"></div>'
  })
  class MyComp {
    constructor(@Attribute('exist') public exist: string) {}
  }

  TestBed.configureTestingModule({declarations: [MyComp]});
  const fixture = TestBed.createComponent(MyComp);
  fixture.detectChanges();
  expect(fixture.componentInstance.exist).toBe('existValue');
});
```
with error:
```
di @Attribute should inject attribute
  Message:
    Expected null to be 'existValue'.
```
Am I missing something obvious?

kara replied:
That test doesn't look quite right. It seems like you're trying to inject the `exist` attribute, but in your test, it's inside the template for `MyComp`, rather than on its host node. I think you need a component wrapping `MyComp` that also has an `exists` attribute. e.g.:

```
template: '<my-comp exist="existValue"></my-comp>'
```

cexbrayat replied:
Argh, of course, sorry...

cexbrayat replied:
@kara OK done! I pushed a fixup commit to let your review.
Would it be useful to migrate all the other `@Attribute` tests as well?

kara replied:
Yes, that would be super helpful! But perhaps in a separate PR?

kara reviewed (changes requested):
Thanks for the PR! Change itself looks great, just needs a test update.

kara reviewed (approved):
LGTM. Just needs a rebase!

kara commented:
[presubmit](http://test/OCL:238132911:BASE:238132962:1552436925034:a3001251)

cexbrayat committed:
fix(ivy): do not inject attributes with namespace

When injecting with `@Attribute`, namespaced attributes should not match (in order to have feature parity with View Engine).

This PR resolves FW-1137

cexbrayat commented:
@kara I rebased 👍 Note that I updated the test to make it look like the one added in #29192 I also wrapped them in a test suite, as it was the case in `render3/di_spec.ts`.

I'll start working on migrating the existing tests in DI when this is merged. Let me know if I need to assign myself on a Jira ticket (I didn't find one, but maybe I missed it).
